aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -102,17 +104,14 @@ IntArrayRef q_per_channel_axis_quant(const Tensor& self) {
 
 Tensor int_repr_quant(const Tensor& self) {
   Tensor dst;
-  // TODO: replace with TensorIterator
-  auto self_c = self.contiguous();
   AT_DISPATCH_QINT_TYPES(self.scalar_type(), "int_repr", [&]() {
     dst = at::empty(self.sizes(), self.options().dtype(UNDERLYING_TYPE));
-    underlying_t* self_data =
-        reinterpret_cast<underlying_t*>(self_c.data_ptr<scalar_t>());
-    underlying_t* dst_data = dst.data_ptr<underlying_t>();
-    if (self.numel() > 0) {
-      memcpy(dst_data, self_data, self.nbytes());
-    }
+    auto iter = TensorIterator::unary_op(dst, self);
+    cpu_kernel(iter, [&](scalar_t value) -> underlying_t {
```

Comment:
@VitalyFedyunin what I want to do is to get the internal representation of a quantized value,
e.g. `scalar_t` is `qint8` and `underlying_t` is `int8`, looks like the `TensorIterator` always tries to compute the common type of the input and output type first, so the `promoteType` function is called, and we don't have rules to promote a quantized type into a non-quantized type. so why is type promotion needed here?

Reply:
https://bddppq.github.io/codebrowser/pytorch/pytorch/aten/src/ATen/native/TensorIterator.cpp.html#_ZN2atL19compute_result_typeEN3c108ArrayRefINS_11OperandInfoEEERKT_